src/firefox/index.js

```diff
@@ -50,27 +50,39 @@ export function defaultRemotePortFinder(
   {
     portToTry=REMOTE_PORT,
     connectToFirefox=defaultFirefoxConnector,
-  }: RemotePortFinderParams = {}
+  }: RemotePortFinderParams = {},
+
 ): Promise<number> {
   log.debug(`Checking if remote Firefox port ${portToTry} is available`);
-
-  return connectToFirefox(portToTry)
-    .then((client) => {
-      log.debug(`Remote Firefox port ${portToTry} is in use`);
-      client.disconnect();
-      // TODO: instead of throw an error, pick a new random port until
-      // one of them is available.
-      // https://github.com/mozilla/web-ext/issues/283
-      throw new WebExtError(
-        `Cannot listen on port ${portToTry} because it's in use`);
+  let retriesCount = 10;
+  function portConnectionRetry(){
+    return new Promise((resolve, reject) => {
```

Comment:
There is no need to construct a promise object here. You can just return `connectToFirefox(portToTry)` because this returns a promise. After that change, you can remove your calls to `resolve()` and `reject()`. Replace `resolve(value)` with `return value` and replace `reject(error)` with `throw error`. This will make the promise chain easier to read.